bundles/org.openhab.binding.bigassfan/src/main/java/org/openhab/binding/bigassfan/internal/discovery/BigAssFanDiscoveryService.java

```diff
@@ -115,21 +115,25 @@ protected void stopBackgroundDiscovery() {
         cancelListenerJob();
     }
 
-    private void startListenerJob() {
-        if (listenerJob == null) {
-            terminate = false;
-            logger.debug("Starting discovery listener job in {} seconds", BACKGROUND_DISCOVERY_DELAY);
-            listenerJob = scheduledExecutorService.schedule(listenerRunnable, BACKGROUND_DISCOVERY_DELAY,
+    private synchronized void startListenerJob() {
+        logger.debug("Starting discovery listener job in {} seconds", BACKGROUND_DISCOVERY_DELAY);
+
+        if (this.listenerJob == null) {
+            this.listenerJob = scheduledExecutorService.schedule(listenerRunnable, BACKGROUND_DISCOVERY_DELAY,
                     TimeUnit.SECONDS);
         }
     }
 
-    private void cancelListenerJob() {
-        if (listenerJob != null) {
-            logger.debug("Canceling discovery listener job");
-            listenerJob.cancel(true);
-            terminate = true;
-            listenerJob = null;
+    @SuppressWarnings("null")
+    private synchronized void cancelListenerJob() {
+        logger.debug("Canceling discovery listener job");
+
+        if (this.listenerJob != null) {
+            if (!this.listenerJob.isCancelled()) {
+                this.listenerJob.cancel(true);
+                terminate = true;
+                listenerJob = null;
```

Comment:
```suggestion
                this.listenerJob = null;
```

Reply:
While i'm fine with adding `this`, what is the point? I have seen some PR's last week where a contributor was asked to remove `this` prefix so is i see some individuall preferences here.

Reply:
Sorry, that was not my intention of the comment, which was about nullifying the instance variable, not the local variable. But I mixed it up and was too fast (some prefer `this.job` and `job` and others `job` and `localJob`). So please disregard that comment.

On a side note, I once had an instance variable named `host`. Naming that `localHost` was to me confusing, so I avoided that. :)